dvc/repo/experiments/show.py

```diff
@@ -136,13 +136,14 @@ def show(
             all_branches=all_branches,
             all_tags=all_tags,
             all_commits=all_commits,
-            sha_only=True,
+            sha_only=False,
```

Comment:
Functionality wise, this just reverts to using command line git (via git-python) to get the tag names, which isn't really a fix (since we eventually want to get rid of git-python entirely).

Reply:
Totally right. 

The snippet you posted when guessing above is exactly where I ended up reaching debugging and it is indeed the source of the problem.

I was a little "afraid" of touching the `scm` module for fixing an (at least initially) `exp show` issue, especially given that `get_ref` is used widely across the experiments functionality (beyond `show`) and that I didn't know what `packed-refs` were or the difference between lightweight and annotated tags 😅 

So I added a "fix" that worked on the `experiments/show` level.

I will try to fix it on `scm` level without breaking anything